app/controllers/v0/messages_controller.rb

```diff
@@ -67,8 +66,55 @@ def thread
 
     private
 
+    def get_folder(folder_id)
+      folders = client.get_folder(folder_id)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, folder_id unless folders.present?
+      folders.data.first
+    end
+
+    # If page is not supplied, we are bringing back all results in the set. In that case
+    # adjust the per_page count to the TBD maximum number of messages allowed per MHV call
+    def pagination_params
+      folder_id = params[:folder_id].try(:to_i)
+      folder = get_folder(folder_id)
```

Comment:
why is this method fetching folder again? should that maybe be stored in an instance variable so that you don't need to fetch it more than once? something like `@folder ||= get_folder(folder_id)`